File: python/DFS.py
class Graph:
    def __init__(self, graph):
        self.graph = graph
        self.visited = []

    def DFS(self, node):
        self.visited.append(node)
        print('visited: ', node)

        for key, value in enumerate(self.graph[node]):
            if value == 1 and key not in self.visited:
                self.DFS(key)

graph = [[0, 1, 1, 1, 0, 0, 1], [1, 0, 1, 0, 0, 0, 0], [1, 1, 0, 0, 1, 0, 0], [1, 0, 0, 0, 0, 1, 0], [0, 0, 1, 0, 0, 1, 0], [0, 0, 0, 1, 1, 0, 1], [1, 0, 0, 0, 0, 1, 0]]
g = Graph(graph)
g.DFS(0)

File: php/DFS.php
<?php

class Graph {

	public $graph = array();

	public $visited = array();

	public function __construct($graph){
		$this->graph = $graph;
	}

 	public function DFS($node) {
	 	array_push($this->visited, $node);
	 	echo $node;
	 	foreach($this->graph[$node] as $key => $value) {
	 		if($value == 1 && !in_array($key, $this->visited)) {
	 			$this->DFS($key);
	 		}
 		}
 	}

}

$graph = array(
	      array(0,1,1,1,0,0,1),
	      array(1,0,1,0,0,0,0),
	      array(1,1,0,0,1,0,0),
	      array(1,0,0,0,0,1,0),
	      array(0,0,1,0,0,1,0),
	      array(0,0,0,1,1,0,1),
	      array(1,0,0,0,0,1,0),
	   );
$g = new Graph($graph);
$g->DFS(0);

?>